### ISSUE TITLE
Please, support the SpotBugs as FindBugs successor

### ISSUE THREAD

**Issue by vektory79** [andrepdo/findbugs-idea#265]
It's sad, but the FindBugs is announced as dead: https://mailman.cs.umd.edu/pipermail/findbugs-discuss/2017-September/004383.html

All further development is moved to SpotBugs. It will be fine to switch to project that will be alive.

Thank You.

**Comment by andrepdo** [comment]
good point, thanks. we should migrate. hopefully the api is nearly compatible so we could migrate without much "pain"

https://spotbugs.github.io

> We hope that all users of FindBugs can move to SpotBugs without big problems, see transition > guide in the new manual.

**Comment by retomerz** [comment]
Yes I have followed the discussion on that.
It is planned to bundle the next version of the "FindBugs-IDEA" plugin with SpotBugs.
Furthermore I will contact JetBrains to ask how we can rename the plugin in their repo without messing things up and if it is possible to keep a "FindBugs-IDEA" alias which redirects, for example, to "SpotBugs-IDEA".

**Comment by scanban** [comment]
I've took initial steps on migration . Just migrated to SpotBugs 3.1.1. Build system changed to gradle and gradle-intellij-plugin, Changed plugin name to "SpotBugs-IDEA" and replaced FindBugs to SpotBugs in plugin.xml description text. Tested with supplied java test project. 
Please take a look:
https://github.com/scanban/findbugs-idea/commit/1bab9a9da436964ef82541c21c4a7fca193a79b2

**Comment by retomerz** [comment]
I have already contacted JetBrains how to rebrand/rename the plugin or if they can add some redirection (which is not possible).
Note that it is not really a good idea to rename everything from "FindBugs" to "SpotBugs":
- XMLID must be the same otherwise the user gets no update notification because its a new plugin
- loose (persistent) plugin settings (all user have to re-configure the plugin settings)

The proposal is:

1. (Optional - but I think this would really make sense)
Fix these bugs:
  https://github.com/andrepdo/findbugs-idea/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+Wrong+line
  https://github.com/andrepdo/findbugs-idea/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+path+in+TreePath
  https://github.com/andrepdo/findbugs-idea/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+Already+disposed

2. Create a new fork/repo here: https://github.com/spotbugs
(We should contact the SpotBugs ppl for that first what they think: @h3xstream, @iloveeclipse, @jsotuyod and @KengoTODA)
And then:
  a. Apply the changes from @scanban
  b. Add some legacy settings converter support which loads the "old" FindBugs settings (so that the user can switch from FindBugs to SpotBugs without re-configure everything. Example:  https://github.com/andrepdo/findbugs-idea/commit/59c8e8f0afa3a38983ffc44b6058bb3a0bb12ab7 )
  c. Perform a plugin release
  
3. Create one last release of FindBugs-IDEA with a proper plugin description that this FindBugs plugin is deprecated and that you should download the new "SpotBugs" plugin instead (with download links).

Please let me know what your thoughts/comments are on this.

**Comment by planetf1** [comment]
Is there any update on spotbugs plugin for INtelliJ?

**Comment by tedyoung** [comment]
Is there anything folks can do (or try out) to help move this along?

**Comment by KengoTODA** [comment]
@tedyoung you may vote for https://youtrack.jetbrains.com/issue/IDEA-201846

**Comment by tedyoung** [comment]
@KengoTODA I did vote, but since this FindBugs-IDEA plugin isn't owned by JetBrains, I'm not sure why they should be responsible for maintaining/upgrading it?

**Comment by KengoTODA** [comment]
Check discussion at https://github.com/spotbugs/spotbugs/issues/515

Simply speaking nobody can contrute to realize it so someone asked them to make official support.

**Comment by MCMicS** [comment]
@retomerz I like your proposal. A fork/repo in spotbugs sounds good.

I've not found any large problems if switch to sportbugs. Only the cloud package is removed and so some changes will be needed in fork or new repo.

**Comment by MCMicS** [comment]
any progress here?

**Comment by MCMicS** [comment]
@retomerz You wrote 

> It is planned to bundle the next version of the "FindBugs-IDEA" plugin with SpotBugs.

Is there any plan for this release?
Then in plugin setting you can choose your version (Findbugs 3.0.1 or bundled spotbugs version).

Or is there any progress with the fork.

**Comment by amaembo** [comment]
The SpotBugs plugin fork is published in the plugin repository
https://plugins.jetbrains.com/plugin/14014-spotbugs
https://github.com/JetBrains/spotbugs-intellij-plugin
